pyazul/validate.py

```diff
@@ -15,32 +63,47 @@ def datavault_create(data):
 def datavault_delete(data):
     required = {
         'DataVaultToken': data.get('DataVaultToken', ''),
-        'TrxType': 'DELETE',
+        'TrxType': r'DELETE',
     }
 
     return required
 
 
 def sale_transaction(data):
-    required = {
-        'CardNumber': data.get('CardNumber', ''),
-        'Expiration': data.get('Expiration', ''),
-        'CVC': data.get('CVC', ''),
-        'PosInputMode': data.get('PosInputMode', 'E-Commerce'),
-        'TrxType': 'Sale',
-        'Payments': data.get('Payments', '1'),
-        'Plan': data.get('Plan', '0'),
-        'Amount': str(utils.clean_amount(data.get('Amount', 0))),
-        'Itbis': utils.clean_amount(data.get('Itbis', 0)),
-        'CurrencyPosCode': data.get('CurrencyPosCode', ''),
-        'AcquirerRefData': data.get('AcquirerRefData', '1'),
-        'CustomerServicePhone': data.get('CustomerServicePhone', ''),
-        'OrderNumber': data.get('OrderNumber', ''),
-        'EcommerceURL': data.get('EcommerceURL', ''),
-        'CustomOrderID': data.get('CustomOrderID', ''),
+    validation_rule_sets = {
+        'CardNumber': (check_exists_in(data), check_not_empty,),
```

Comment:
I feel like there's some DRY improvement that could be done here with `check_exists_in()`. I'll analyze the possibilities further.

Reply:
please do :D.
Since the original name for the dic was "required" I just assumed all of those are required. Maybe we can't remove it from a few.

Also `check_exists_in` has two proposes, it allows to specify a field (key or value) that must be within a set of options.

E.g. `check_exist_in(["optionA", "OptionB", "OptionsC"])` Some renaming might clarify this second functionality.

Reply:
if we identify fields that aren't required this line could be changed to update accordingly and skip the other checks if it isn't present:

https://github.com/indexa-git/pyazul/pull/8/files#diff-0bad0c540736287f54cc0b0182b7daffR45